pruntime/enclave/src/contracts/balance.rs

```diff
@@ -94,6 +123,36 @@ impl contracts::Contract<Command, Request, Response> for Balance {
                 } else {
                     TransactionStatus::NoBalance
                 }
+            },
+            Command::TransferToChain {dest, value} => {
+                let o = AccountIdWrapper(origin.clone());
+                println!("Transfer to chain: [{}] -> [{}]: {}", o.to_string(), dest.to_string(), value);
+                if let Some(src_amount) = self.accounts.get_mut(&o) {
+                    if *src_amount >= value {
+                        let src0 = *src_amount;
+                        *src_amount -= value;
+                        println!("   src: {:>20} -> {:>20}", src0, src0 - value);
+
+                        //push to queue
+                        let sequence = self.sequence.clone() + 1;
+                        let transfer_data = TransferData {
+                            id: (rand::random::<u32>(), &dest, sequence).using_encoded(blake2_256),
```

Comment:
Better to use the extrinsic id as the id here, and btw, avoid non-deterministic in any contract code